uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/manager-leave-calendar.js

```diff
@@ -316,39 +210,114 @@ define([
-       * @param  {object} leaveRequest
-       */
-      function deleteLeaveRequest (event, leaveRequest) {
-        var contactID = leaveRequest.contact_id;
-        // Find the calendarData for the deleted requests user
-        var calendar = _.find(calendarData, function (calendar) {
-          return calendar.contact_id === contactID;
-        });
+    /**
+     * Event handler for Delete event of Leave Request
+     *
+     * @param  {object} event
+     * @param  {LeaveRequestInstance} leaveRequest
+     */
+    function deleteLeaveRequest (event, leaveRequest) {
+      var contactID = leaveRequest.contact_id;
+      // Find the calendarData for the deleted requests user
+      var calendar = _.find(calendarData, function (calendar) {
+        return calendar.contact_id === contactID;
+      });
 
-        vm.leaveRequests[contactID] = _.omit(vm.leaveRequests[contactID], function (leaveRequestObj) {
-          return leaveRequestObj.id === leaveRequest.id;
+      vm.leaveRequests[contactID] = _.omit(vm.leaveRequests[contactID], function (leaveRequestObj) {
+        return leaveRequestObj.id === leaveRequest.id;
+      });
+      vm._resetMonths();
+      vm._setCalendarProps(contactID, calendar);
+    }
+
+    /**
+     * Load all contacts with respect to filters
+     *
+     * @return {Promise}
+     */
+    function loadContacts () {
+      return Contact.all(prepareContactFilters(), null, 'display_name')
+        .then(function (contacts) {
+          vm.filteredContacts = contacts.list;
         });
-        vm._resetMonths();
-        vm._setCalendarProps(contactID, calendar);
+    }
+
+    /**
+     * Loads the managees of currently logged in user
+     *
+     * @return {Promise}
+     */
+    function loadManagees () {
+      if (isAdmin) {
+        return Contact.all()
+          .then(function (contacts) {
+            vm.managedContacts = contacts.list;
+            vm.filteredContacts = contacts.list;
+          });
       }
-    })();
+
+      return Contact.find(vm.contactId)
+        .then(function (contact) {
+          return contact.leaveManagees();
+        })
+        .then(function (contacts) {
+          vm.managedContacts = contacts;
+        })
+        .then(function () {
+          return loadContacts();
+        });
+    }
+
+    /**
+     * Loads the OptionValues necessary for the controller
+     *
+     * @return {Promise}
+     */
+    function loadOptionValues () {
+      return OptionGroup.valuesOf([
+        'hrjc_region',
+        'hrjc_location',
+        'hrjc_level_type',
+        'hrjc_department'
+      ])
+      .then(function (data) {
+        vm.regions = data.hrjc_region;
+        vm.locations = data.hrjc_location;
+        vm.levelTypes = data.hrjc_level_type;
+        vm.departments = data.hrjc_department;
+      });
+    }
+
+    /**
+     * Returns the filter object for contacts api
+     *
+     * @return {Object}
+     */
+    function prepareContactFilters () {
+      return {
```

Comment:
**var filters = vm.filters;**
...
**filters**.department ? **filters**.department.value : **undefined**

I think it is safer to send `undefined` as it means the filter should be completely _ignored_, but not that the filter's value is NULL